### PR TITLE
Use `gitlab-api` plugin

### PR DESCRIPTION
**Pull request by basil**
Using the new `gitlab-api` plugin from https://github.com/jenkinsci/gitlab-api-plugin/pull/7

It took a while to get this compile. I have not tried to run the code and I do not have a GitLab server to test with.

Anyone interested in testing this? @zbynek perhaps since you submitted some security fixes recently (no pressure, though)? @dcendents since you're doing other GitLab plugin stuff?

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -82,32 +82,9 @@
...
-
-                <!-- Provided by Jenkins core -->
-                <exclusion>
-                    <groupId>org.slf4j</groupId>
-                    <artifactId>slf4j-api</artifactId>
-                </exclusion>
-            </exclusions>
+            <groupId>io.jenkins.plugins</groupId>
+            <artifactId>gitlab-api</artifactId>
+            <version>1.0.7-rc68.a_3711d5d3b_1c</version> <!-- TODO https://github.com/jenkinsci/gitlab-api-plugin/pull/7 -->
```
Comment (jetersen) [inline pom.xml:87]:
```suggestion
            <version>5.0.1-69.vcdc19e779072</version>
```

**Comment by dcendents** [comment]
@basil I don't use this plugin so it is not easy for me to test either.

**Comment by basil** [comment]
Released in [1.16](https://github.com/jenkinsci/gitlab-oauth-plugin/releases/tag/gitlab-oauth-1.16).

**Comment by zbynek** [comment]
I installed the RC build from this PR and didn't spot any problems (login/logout worked for me). Will try again with 1.16

**Comment by basil** [comment]
Thanks @zbynek. I installed a toy instance of GitLab this morning and did a basic smoke test, but I don't have a realistic testing environment. Confirmation that this is stable and working would definitely be appreciated!

**Comment by zbynek** [comment]
Still works fine with 1.16 :+1:

**Pull request merged**